mutatex/utils.py

```diff
@@ -410,65 +410,80 @@ def get_foldx_sequence(pdb, multimers=True):
     except:
         log.error("couldn't read or parse your PDB file")
         raise IOError
 
+    #identify the models and any inconsistencies
+    models = list(structure.get_models())
+    if len(models) > 1:
+        log.warning("%d models are present in the input PDB file; only the first will be used." % len(models))
+    if len(models) < 1:
+        log.error("The input PDB file does not contain any model. Exiting ...")
+        raise IOError
+
+    #define the model from models, as len(models) should be == 1.
+    model = models[0]
+
+    #define empty list for residues (used for multimers) 
     residue_list = []
-    sequences = {}
-    positions = {}
+    #define empty dictionary for sequences (the amino acid sequence)
+    sequences = defaultdict(str)
+    #define empty dictionary for amino acid, position number, chain letter
+    chain_residues = defaultdict(list)
 
-    for model in structure:
-        for chain in model:
-            chain_name = chain.get_id()
-            sequences[chain_name] = ''
-            positions[chain_name] = ''
-            for residue in chain:
-                try:
-                    res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
-                except:
-                    log.warning("Residue %s in file %s couldn't be recognized; it will be skipped" %(residue, pdb))
-                    continue
-                if not multimers:
-                    residue_list.append(tuple(["%s%s%d" % (res_code, chain.get_id(), residue.get_id()[1])]))
-                else:
-                    sequences[chain_name] += res_code
-                    positions[chain_name] += f"{residue.get_id()[1]},"
+    # Extract residues and sequences from chains
+    for chain in model:
+        chain_name = chain.get_id()
+        for residue in chain:
+            try:
+                res_code = PDB.Polypeptide.three_to_one(residue.get_resname())
+            except KeyError:
+                log.warning("Residue %s couldn't be recognized; it will be skipped" % residue)
+                continue
+
+            resid = residue.get_id()[1] #the position number
+            sequences[chain_name] += res_code #the one letter amino acid
+            chain_residues[chain_name].append((res_code, resid, chain_name)) #one letter amino acid, position number, chain letter
 
     if multimers:
-        collated_chains = []
-        seq_ids, seqs = list(zip(*list(iteritems(sequences))))
-        pos_ids, pos = list(zip(*list(iteritems(positions))))
-        seq_ids = np.array(seq_ids)
-        pos_ids = np.array(pos_ids)
+        #combine sequences and residue positions
+        chain_names = list(sequences.keys())
+    
+        #combined identifier: sequence + residue numbering
+        #use string to overcome np.array challenges with length when the numbers mismatch.
+        chain_residue_patterns = [
+            str(tuple((res_code, resid) for res_code, resid, chain in chain_residues[chain_name]))
```

Comment:
using `str` here is a bit of a hack and not very straightforward - maybe we can find a slightly more robust way to handle this, but I didn't quite get what the problem is. 

One idea could be using `f"{resid}_{res_code}" instead of building a tuple here, then it should work. The `_` is only to leave a placeholder for chain name